e2e-tests/page-objects/BasePage.ts

```diff
@@ -0,0 +1,46 @@
+import { test, expect, type Page, type Browser, type BrowserContext } from '@playwright/test';
+import BasePageElement from './BasePageElement';
+
+export type BasePageConstructor = {
+    browser: Browser;
+    baseUrl?: string;
+}
+
+export interface BasePageInterface  {
+    open: () => Promise<void>
+    init: () => Promise<void>
+}
+
+export default class BasePage implements BasePageInterface {
+    public page: Page;
+    public baseUrl: string;
+    public browser: Browser;
+    private context: BrowserContext;
+
+    constructor({ browser, baseUrl }: BasePageConstructor) {
+        this.baseUrl = baseUrl ?? '';
+        this.browser = browser;
+    }
+
+    protected async initContext() {
+        this.context = await this.browser.newContext();
+        this.page = await this.context.newPage();
```

Comment:
Should this only ever run once per page? Should we guard double executions with a flag?

Reply:
Yes, makes sense. Initially, I've added a separate init method that would create the page context, but I forgot to call it a few times (and got undefined in runtime :D) , so having the context creation in the open method was reasonable. Will improve in the next PR :)